### ISSUE TITLE
Benchmarks for SSA/jump systems

### ISSUE THREAD

**Issue by isaacsas** [JuliaDiffEq/DiffEqBenchmarks.jl#25]
Before starting any benchmarking, it's probably a good idea to think about what we want to look at. Some ideas:

1. Varying number of species, number of reactions, and network topologies (i.e. sparsity). Many SSAs are designed to scale well in one of number of species or number of reactions, but may not scale well as both are increased. Many SSAs are designed to exploit sparsity in network structure, but may be slower than the basic direct method for dense networks.

2. Increasing simulation stopping time and/or species population sizes. Both lead to an increase in the number of jumps (i.e. events) within one simulation. 

3. Separation vs. clustering of jump rates. Some SSAs work very well for systems with large timescale separations (i.e. they are optimized to quickly handle frequent jumps). Systems with many identical jump rates (like diffusion systems) may cause them issues.

Testing "accuracy" seems less relevant for exact SSAs; this would really just be looking at sampling error, which should converge the same for all exact methods...

**Comment by ChrisRackauckas** [comment]
>Testing "accuracy" seems less relevant for exact SSAs; this would really just be looking at sampling error, which should converge the same for all exact methods...

I agree, though testing tau-leaping will be interesting. For exact, raw timing numbers, or their change as the complexity is varied is all that's needed. So not much tooling other than example problems is necessary.
